### PR TITLE
Init basic structure

### PR DESCRIPTION
**Pull request by wachterjohannes**
This PR is used to build the basic structure of the new sulu-edition.

Fixes sulu/sulu#2406

Depends on sulu/sulu#2408
#### TODO
- [x] console for admin and website (makes problems because of distribution-bundle)
- [ ] move sulu.yml files (maybe into the bundles or a basic yml in core-bundle) (will be fixed in own PR #2)
- [ ] move routing into basic xml file in admin-bundle (what will we do for optional bundles?) (will be fixed in own issue #4)
- [ ] navigation omit the port on the website (https://github.com/sulu/sulu/issues/2423)
- [x] twig-files with javascript and css

### CODE REVIEW

**Comment by wachterjohannes** [comment]
i have not fixed the console problem. but i have introduced an `./bin/adminconsole` and a `./bin/websiteconsole`. if you are using `./bin/console` you will get a warning that there are this two other consoles for the contexts.

File: bin/console [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+use Symfony\Component\Console\Output\ConsoleOutput;
+use Symfony\Component\Console\Style\SymfonyStyle;
+
+require_once __DIR__ . '/../app/autoload.php';
+
+$input = new ArgvInput();
+$output = new ConsoleOutput();
+$io = new SymfonyStyle($input, $output);
+
+$io->warning('This command will be executed in the context "admin". If you want to use another context please use "./bin/adminconsole" or "./bin/websiteconsole".');
```
Comment (wachterjohannes) [inline bin/console]: we should talk about this text. i am quite sure that we can explain that better (:

File: app/Resources/templates/pages/default.xml [hunk 2]
```diff
@@ -0,0 +1,50 @@
+<?xml version="1.0" ?>
+<template xmlns="http://schemas.sulu.io/template/template"
+          xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+          xsi:schemaLocation="http://schemas.sulu.io/template/template http://schemas.sulu.io/template/template-1.0.xsd">
+
+    <key>default</key>
+
+    <view>:templates:/default</view>
```
Comment (webmozart) [inline app/Resources/templates/pages/default.xml]: It should be possible to write `templates/default` here.

File: app/Resources/templates/pages/homepage.xml [hunk 3]
```diff
@@ -0,0 +1,50 @@
+<?xml version="1.0" ?>
+<template xmlns="http://schemas.sulu.io/template/template"
+          xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+          xsi:schemaLocation="http://schemas.sulu.io/template/template http://schemas.sulu.io/template/template-1.0.xsd">
+
+    <key>default</key>
+
+    <view>:templates:homepage</view>
```
Comment (webmozart) [inline app/Resources/templates/pages/homepage.xml]: as above

File: app/Resources/templates/pages/default.xml [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+        <title lang="en">Default</title>
+    </meta>
+
+    <properties>
+        <section name="highlight">
+            <properties>
+                <property name="title" type="text_line" mandatory="true">
+                    <meta>
+                        <title lang="de">Titel</title>
+                        <title lang="en">Title</title>
```
Comment (webmozart) [inline app/Resources/templates/pages/default.xml]: I'd list "en" as topmost locale everywhere.

File: app/Resources/views/templates/default.html.twig [hunk 5]
```diff
@@ -0,0 +1,9 @@
+{% extends "::master.html.twig" %}
```
Comment (webmozart) [inline app/Resources/views/templates/default.html.twig]: `master.html.twig`

File: app/Resources/views/templates/homepage.html.twig [hunk 6]
```diff
@@ -0,0 +1,9 @@
+{% extends "::master.html.twig" %}
```
Comment (webmozart) [inline app/Resources/views/templates/homepage.html.twig]: as above

File: app/Resources/webspaces/example.com.xml [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+        <localization language="en" default="true"/>
+    </localizations>
+
+    <default-templates>
+        <default-template type="page">default</default-template>
+        <default-template type="home">homepage</default-template>
+    </default-templates>
+
+    <error-templates>
+        <error-template default="true">:error:default.html.twig</error-template>
```
Comment (webmozart) [inline app/Resources/webspaces/example.com.xml]: as above

File: app/config/sulu.yml [hunk 8]
```diff
@@ -0,0 +1,150 @@
...
+            use_page_ttl: true
+            enabled: true
+        debug:
+            enabled: %kernel.debug%
+    proxy_client:
+        symfony:
+            enabled: true
+
+dubture_f_fmpeg:
+    ffmpeg_binary: /usr/local/bin/ffmpeg # path to ffmpeg
```
Comment (webmozart) [inline app/config/sulu.yml]: We should try to make this OS independent.

File: composer.json [hunk 9]
```diff
@@ -0,0 +1,86 @@
...
+        "classmap": [
+            "app/AbstractKernel.php",
+            "app/AdminKernel.php",
+            "app/WebsiteCache.php",
+            "app/WebsiteKernel.php"
+        ]
+    },
+    "require": {
+        "php": ">=5.5",
+        "twig/extensions": "~1.0",
```
Comment (webmozart) [inline composer.json]: I'd use the notation `^1.0` here and everywhere else in this file.

File: var/.gitignore [hunk 10]
```diff
@@ -0,0 +1,11 @@
+cache/*
```
Comment (webmozart) [inline var/.gitignore]: I would merge the .gitignore files with the toplevel one.

File: web/website.php [hunk 11]
```diff
@@ -0,0 +1,54 @@
...
+if (SYMFONY_DEBUG) {
+    Debug::enable();
+}
+
+$kernel = new WebsiteKernel(SYMFONY_ENV, SYMFONY_DEBUG);
+$kernel->loadClassCache();
+
+// Comment this line if you want to use the "varnish" http
+// caching strategy. See http://sulu.readthedocs.org/en/latest/cookbook/caching-with-varnish.html
+if (SYMFONY_ENV != 'dev') {
```
Comment (webmozart) [inline web/website.php]: `!==`

**Comment by webmozart** [comment]
Looks good to me apart from my comments.

File: app/config/sulu.yml (hunk 8 above)
Comment (wachterjohannes) [inline app/config/sulu.yml]: i have created an issue in the library and in this repository #3

**Comment by wachterjohannes** [comment]
@webmozart i have fixed everything

**Comment by wachterjohannes** [comment]
@webmozart could you also have a look at the warning message https://github.com/sulu/sulu-minimal/pull/1#discussion_r66402180

File: bin/console (hunk 1 above)
Comment (webmozart) [inline bin/console]: This command will be executed in the "admin" context. For the "website" context, run ./bin/websiteconsole

**Comment by webmozart** [comment]
See above :)

File: bin/console (hunk 1 above)
Comment (wachterjohannes) [inline bin/console]: fixed

File: web/website.php [hunk 12]
```diff
@@ -0,0 +1,54 @@
...
+/*
+ * This file is part of Sulu.
+ *
+ * (c) MASSIVE ART WebServices GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+use Symfony\Component\ClassLoader\ApcClassLoader;
```
Comment (chirimoya) [inline web/website.php]: remove use

File: web/admin.php [hunk 13]
```diff
@@ -0,0 +1,44 @@
...
+/*
+ * This file is part of Sulu.
+ *
+ * (c) MASSIVE ART WebServices GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+use Symfony\Component\ClassLoader\ApcClassLoader;
```
Comment (chirimoya) [inline web/admin.php]: remove use

File: web/admin.php [hunk 14]
```diff
@@ -0,0 +1,44 @@
...
+$loader = require __DIR__ . '/../app/autoload.php';
+include_once __DIR__ . '/bootstrap.php.cache';
+
+if (SYMFONY_DEBUG) {
+    Debug::enable();
+}
+
+$kernel = new AdminKernel(SYMFONY_ENV, SYMFONY_DEBUG);
+$kernel->loadClassCache();
+
```
Comment (chirimoya) [inline web/admin.php]: remove empty line

**Pull request merged**